mobgap/sl/_stride_length.py

```diff
@@ -0,0 +1,296 @@
+        A cross-validation on speed ranges." IEEE TNSRE, vol 29, pp. 1955-1964, 2021.
+    .. [3] https://github.com/mobilise-d/Mobilise-D-TVS-Recommended-Algorithms/blob/master/SLA_SLB/STRIDELEN.m
+
+    """
+
+    orientation_method: MadgwickAHRS
+    step_length_smoothing: BaseFilter
+    max_interpolation_gap_s: int
+    class PredefinedParameters:
+        """
+        The tuning coefficients to be used in the biomechanical model.
+        The reported parameters were previously accessed with model.K.ziljsV3.
+        Coefficients are divided by four to separate the coefficient from
+        the conversion from step length to stride length and from the x2 factor of the formula declared
+        in [1, 2].
+        """
+        tuning_coefficient_MS_MS = {"tuning_coefficient": 4.587/4, "tuning_cohort": "MS_MS"}
+        tuning_coefficient_MS_ALL = {"tuning_coefficient": 4.739/4, "tuning_cohort": "MS_ALL"}
+    @set_defaults(**PredefinedParameters.tuning_coefficient_MS_MS)
+    def __init__(
+        self,
+        *,
+        orientation_method: MadgwickAHRS = None,
+        step_length_smoothing: BaseFilter = cf(HampelFilter(2, 3.0)),
+        max_interpolation_gap_s: int = 3,
+        tuning_coefficient: float,
+        tuning_cohort: str,
+    ) -> None:
+        self.step_length_smoothing = step_length_smoothing  # TODO: understand how to pass step_length_smoothing to _stride2sec
+        self.max_interpolation_gap_s = max_interpolation_gap_s
+        self.orientation_method = orientation_method  # TODO: understand how to optionally perform rotation
+        self.tuning_coefficient = tuning_coefficient
+        self.tuning_cohort = tuning_cohort
+
+    @sl_docfiller
+    def calculate(
+        self,
+        data: pd.DataFrame,
+        *,
+        initial_contacts: pd.Series,
+        sampling_rate_hz: float,
+        sensor_height: float,
+        **_: Unpack[dict[str, Any]],  # TODO: which default value for orientation method?
+    ) -> Self:
+        """%(calculate_short)s.
+
+        Parameters
+        ----------
+        %(calculate_para)s
+
+        %(calculate_return)s
+
+        """
+        self.data = data
+        self.initial_contacts = initial_contacts
+        self.sampling_rate_hz = sampling_rate_hz
+        self.sensor_height = sensor_height
+
+        # 1. Sensor alignment (optional): Madgwick complementary filter
+        newAcc = data[["acc_x", "acc_y", "acc_z"]]  # consider acceleration
+        if self.orientation_method != None:  # TODO: how to make rotation optional?
+            # perform rotation
+            rotated_data = rotate_dataset_series(
+                data, self.orientation_method.estimate(data, sampling_rate_hz=sampling_rate_hz).orientation_object_[:-1]
+            )
+            newAcc = rotated_data[["acc_x", "acc_y", "acc_z"]]  # consider acceleration
+            print("Reorientation performed!")
```

Comment:
We should remove print statements in the final version

Reply:
Done!